docs/pages/architecture/agents.mdx

```diff
@@ -0,0 +1,265 @@
+
+</Details>
+
+## Connecting to infrastructure through Teleport agents
+
+Teleport users can access resources in their infrastructure with the `tsh`
+clientCLI, the Teleport Web UI, and Teleport Connect. All of these tools route
+traffic through agents, usually routed through the Teleport Proxy Service.
+
+### Communication between agents and infrastructure resources
+
+Teleport agents proxy user traffic to resources in your infrastructure. The
+method that agents use to connect to a resource typically depends on whether the
+resource is self hosted or managed by a cloud provider. 
+
+For self-hosted resources, operators distribute mutual TLS credentials to
+infrastructure resources, including a certificate signed by a Teleport
+certificate authority for validating the resource and a certificate authority
+certificate that the resource can use for validating Teleport-issued client
+certificates. When an agent connects to a resource, it retrieves a signed client
+certificate from the Teleport Auth Service.
+
+For resources managed by cloud providers, Teleport takes advantage of identity
+and access management (IAM) solutions available through the cloud provider. The
+agent accesses credentials to authenticate to the cloud provider, and has
+permissions to perform certain operations against the infrastructure resources
+it protects, e.g., managing a database or assuming another IAM role.
+
+To learn more about the mechanism an agent uses to authenticate to an
+infrastructure resource, read the guide to setting up connectivity to that
+resource:
+
+- [Applications](../application-access/guides.mdx)
+- [Cloud provider APIs](../application-access/cloud-apis.mdx)
+- [Databases](../database-access/guides.mdx)
+- [Kubernetes clusters](../kubernetes-access/register-clusters.mdx)
+- [Linux hosts with Teleport](../server-access/getting-started.mdx)
+- [OpenSSH servers](../server-access/openssh.mdx)
+- [Windows desktops](../desktop-access/gertting-started.mdx)
+
+### Communication between clients and agents
+
+Teleport makes several client tools available for access infrastructure
+resources through agents:
+
+- [The `tsh` CLI](../connect-your-client/tsh.mdx)
+- [Teleport Connect](../connect-your-client/teleport-connect.mdx)
+- [Teleport Web UI](../connect-your-client/web-ui.mdx)
+
+After retrieving [client credentials](#credentials-for-teleport-clients), these
+tools authenticate to the Teleport Proxy Service and send traffic to Teleport
+agents. The protocol used depends depends on the upstream resource. See [TLS
+Routing](./tls-routing.mdx) for how client tools perform protocol negotiation
+with the Teleport Proxy Service.
+
+Depending on the upstream infrastructure resource a user wants to connect to,
+the `tsh` client tool may spin up one or more local proxy servers. These
+authenticate to the Teleport Proxy Service, then create a listener that accepts
+traffic from local client tools. A user can then connect to the local proxy with
```

Comment:
maybe an example or two to prevent the user from getting lost in the generalities here
```suggestion
traffic from local, non-Teleport client tools (such as `psql` or `kubectl`). A user can then connect to the local proxy with
```